estate/models/offer.py

```diff
@@ -0,0 +1,85 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+from datetime import timedelta
+from odoo.tools.float_utils import float_compare
+
+
+class Offer(models.Model):
+    _name = "estate.offers"
+    _description = "Offers"
+
+    price = fields.Integer(required=True)
+    status = fields.Selection(
+        [("accepted", "Accepted"), ("refused", "Refused")],
+        string="Status",
+        required=False,
+    )
+    building_id = fields.Many2one("estate.buildings", string="Building")
+    partner_id = fields.Many2one("res.partner", string="Partner")
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(
+        string="Deadline",
+        compute="_compute_date_deadline",
+        inverse="_inverse_date_deadline",
+    )
+
+    _price_positive_constraint = models.Constraint(
+        "CHECK (price > 0)", "Offer price must be positive."
+    )
+
+    @api.depends("validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = fields.Date.today() + timedelta(days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - fields.Date.today()).days
+
+    def action_accept_offer(self):
+        for record in self:
+            if record.status != "accepted" and record.building_id.state not in [
+                "sold",
+                "canceled",
+            ]:
```

Comment:
IMHO this is more readable
```suggestion
            if record.status != 'accepted' and record.building_id.state not in ['sold', 'canceled']:
```